### PR TITLE
Fix for audio deadlock & audio device selection

### PR DESCRIPTION
**Pull request by jherico**
Because of a VS 2013 bug in `std::call_once`, the use of a blocking connections on the main thread to request the `AudioClient` to do something is causing a deadlock.  This is because the main thread is inside a `std::call_once` and the AudioClient _may_ enter into another `std::call_once` before it processes the blocking invocation.  

## Testing

If you're experiencing deadlocks on startup with build 6896, please test with this build and see if they still occur.

To test further...  

  * remove any `Audio/XXX/INPUT` entries from your Interface.json file (located in `~\AppData\Roaming\High Fidelity\` or `~\AppData\Roaming\High Fidelity - dev` for PR builds
 * Open Interface
 * In desktop mode, the audio input and output should match the defaults in your system control panel
 * In Oculus mode the audio inputs and output should automatically switch to the appropriate Rift default devices
 * In either mode, if you change the settings and exit interface, when you restart and go back into that mode (desktop or HMD) the settings should be preserved.

### CODE REVIEW

**Comment by Nex-Pro** [comment]
Will test when built

**Comment by Nex-Pro** [comment]
@jherico  can I assume oculus mode also applies to Vive mode ?

**Comment by jherico** [comment]
@Nex-Pro the way OpenVR handles audio devices is different from the Oculus approach.  When you activate OpenVR it will optionally change your system default audio device (based on your preferences in the OpenVR control panel).  So the Vive doesn't really expose a preferred audio device in the same way as the Oculus.  I'm not sure what the interaction would be.

**Comment by Nex-Pro** [comment]
@jherico I'll drop this if not needed

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1855/HighFidelity-Beta-PR11023-5b5e94700ecc20e0cfd939bec833e8f14a4e187d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1855/HighFidelity-Beta-PR11023-5b5e94700ecc20e0cfd939bec833e8f14a4e187d.exe

**Comment by Triplelexx** [comment]
Does fix the deadlock I was encountering.

File: interface/src/scripting/AudioDevices.cpp [hunk 1]
```diff
@@ -85,6 +105,7 @@ void AudioDeviceList::resetDevice(bool contextIsHMD, const QString& device) {
             QMetaObject::invokeMethod(client, "switchAudioDevice", Q_ARG(QAudio::Mode, _mode));
         }
     }
+#endif
```
Comment (AndrewMeadows) [inline interface/src/scripting/AudioDevices.cpp]: Is this cruft?  or important reference implementation?

**Pull request merged**